src/Microsoft.Health.Dicom.Api/Web/AspNetCoreMultipartReader.cs

```diff
@@ -101,6 +102,10 @@ public async Task<MultipartBodyPart> ReadNextBodyPartAsync(CancellationToken can
         {
             throw new InvalidMultipartRequestException(ex.Message);
         }
+        catch (BadHttpRequestException)
```

Comment:
Do you know if there are any other reasons the `ReadNextSectionAsync` could throw a `BadHttpRequestException`?

Reply:
There areactually a lot: https://github.com/dotnet/aspnetcore/blob/main/src/Servers/Kestrel/Core/src/Internal/Http/Http1ContentLengthMessageBody.cs

I've updated the approach to look for the specific message, which unfortunately is not exposed as a constant by ASP Net Core.